enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,462 @@
+}
+
+//AddOnPlacementScoreStatus represents the current status of AddOnPlacementScore.
+type AddOnPlacementScoreStatus struct {
+	// Conditions contain the different condition statuses for this AddOnPlacementScore.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// Scores contains a list of score name and value of this managed cluster.
+	// +optional
+	Scores []AddOnPlacementScoreItem `json:"scores,omitempty"`
+
+	// ValidUntil defines the valid time of the scores.
+	// After this time, the scores are considered to be invalid by placement. nil means never expire.
+	// The controller owning this resource should keep the scores up-to-date.
+	// +optional
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	ValidUntil *metav1.Time `json:"validUntil" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+}
+
+//AddOnPlacementScoreItem represents the score name and value.
+type AddOnPlacementScoreItem struct {
+	// Name is the name of the score
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// Value is the value of the score. The score range is from -100 to 100.
+	// +kubebuilder:validation:Minimum:=-100
+	// +kubebuilder:validation:Maximum:=100
+	// +required
+	Value int32 `json:"value,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AddOnPlacementScoreList is a collection of AddOnPlacementScore.
+type AddOnPlacementScoreList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// Items is a list of AddOnPlacementScore
+	Items []AddOnPlacementScore `json:"items"`
+}
+```
+### Placement API
+Replace `Name` with `PrioritizerScoreCoordinate` inside `PrioritizerConfig`, to support specifying the customized score to sort clusters.
+
+```golang
+type Placement struct {
+  ...
+	Spec PlacementSpec `json:"spec"`
+}
+
+type PlacementSpec struct {
+  ...
+
+	// PrioritizerPolicy defines the policy of the prioritizers.
+	// If this field is unset, then default prioritizer mode and configurations are used.
+	// Referring to PrioritizerPolicy to see more description about Mode and Configurations.
+	// +optional
+	PrioritizerPolicy PrioritizerPolicy `json:"prioritizerPolicy"`
+}
+
+// PrioritizerPolicy represents the policy of prioritizer
+type PrioritizerPolicy struct {
+  ...
+
+	// +optional
+	Configurations []PrioritizerConfig `json:"configurations,omitempty"`
+}
+
+// PrioritizerConfig represents the configuration of prioritizer
+type PrioritizerConfig struct {
+	// PrioritizerScoreCoordinate represents the configuration of the prioritizer and score source.
+	// +kubebuilder:validation:Required
+	// +required
+	PrioritizerScoreCoordinate PrioritizerScoreCoordinate `json:"scoreCoordinate"`
```

Comment:
this is a proposed change to `Name` in the existing API?

Reply:
Yes, as this is v1alpha1 API, might add `Name` back and claim it will be deprecated.